src/main/java/com/hivemq/extensions/aws/S3Client.java

```diff
@@ -44,22 +46,25 @@
  */
 public class S3Client {
 
-    private static Logger logger = LoggerFactory.getLogger(S3Client.class);
+    private static final Logger logger = LoggerFactory.getLogger(S3Client.class);
 
-    private S3Config s3Config;
-    public ConfigurationReader configurationReader;
-    public AmazonS3 amazonS3;
+    public @NotNull ConfigurationReader configurationReader;
```

Comment:
can this be final?

Reply:
Normally it would be final but we need to change the value of it for test purposes.